addons/website/controllers/backend.py

```diff
@@ -55,3 +55,28 @@ def check_create_access_rights(self, models):
             model: request.env[model].check_access_rights('create', raise_exception=False)
             for model in models
         }
+
+    @http.route('/website/track_installing_modules', type='json', auth='user')
+    def website_track_installing_modules(self, selected_features, total_features=None):
+        """
+        During the website configuration, this route allows to track the
+        website features being installed and their dependencies in order to
+        show the progress between installed and yet to install features.
+        """
+        features = request.env['website.configurator.feature'].browse(selected_features)\
+            .filtered(lambda feature: feature.module_id.state != 'installed')
```

Comment:
This part

Reply:
As discussed:
1. For the progressbar to work, you should launch the local server with `--workers 4`.
2. I haven't found a way to directly get those numbers shown in the logs. They are generated on the fly in `odoo/modules/loading.py` (search for "Loading module"), but there doesn't seem to be an easy way to hook up to these from outside. (But if I'm wrong I'll be glad to know!)

Reply:
I don't see how the progressbar could work at all with this implementation.

Let's take the main flow: installing only one app, and probably ecommerce that is.

The current code will be looking for when the `website_sale` module is installed before updating the progressbar.
There is conceptually 2 problems:
1. It means that there is only 2 states in the progress bar: 0% when website_sale is not installed and 100% when it is installed. This means that it will not be a progress bar at all, the goal was (and is) to show a.. **progress** bar, meaning it should show you how is the progress going: 1% 4% 14% 22% 30% etc etc, it should be moving during all the processing time.
The goal of this task is to avoid the user thinking something is stuck and that the page is not working anymore. This will be doing the opposite as this new element on the UI will also look stuck now. 
2. On top of having a limited steps in the progress bar (basically the number of selected apps), all those steps will be reached only after ~90% of the process, because the majority of the processing time is installing dependency apps which are not the selected apps.
For instance, installe `website_sale` will install many many many many apps like accounting and so on, which won't have an impact on the progress bar.

If we can't find a solution, this progress bar can simply be removed from the PR IMHO, it's just something I pushed for on the spec and that was not supposed to be there in the first place, so if it's not really helpful and adding value, we might as well get rid of it (less code).

I will have a look to see if I can find a way to make it somehow work like that.

Reply:
I actually pushed this commit https://github.com/odoo/odoo/pull/118525/commits/2ede242258d27034d5e86eafd5fd78788bbff7be
(Just a crappy wip)

Which basically do:
```py
        a = request.env['website.configurator.feature'].browse(selected_features).module_id.upstream_dependencies(exclude_states=('',))
        features_info = {
            'total': len(a),
            'nbInstalled': len(a.filtered(lambda f: f.state == 'installed')),
        }
        return features_info
```
Which seems to almost behaves as expected:
- It's missing the auto install modules which are not depedencies of any module like web_unsplash, so I think there is a discrepancy between the logs and what we got here (like ~35 vs ~55) because of that -> Maybe to check if can be fixed? Otherwise maybe not the end of the world 
- I think the % will start at like 50% because a good junk of the apps are already installed (base, website, portal etc).
Need to investigate if we can have the list of installed app before applying the configurator, so we know which one to remove from the computation.

Note that I am surprised we have access to those installed module in real time (I mean in a rpc while the module install process is still pending) -> I would have expect this install to be inside a transaction and so nothing commited before the end and so any other cursor not aware of it, but I guess some stuff are commited during the install.